float64 distance
float64 direction